algs/resample/src/row_resample.cu

```diff
@@ -0,0 +1,96 @@
+/**
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 3 of the License, or (at your option)
+ * any later version.
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, see http://www.gnu.org/licenses/
+ */
+
+#include "row_resample.h"
+
+#include <cmath>
+
+#include "cuda_ptr.h"
+#include "cuda_util.h"
+
+namespace {
+
+__host__ __device__ inline void InterpolatePixelValue(int x, const float* in, int in_size, float* out, int out_size,
+                                                      double pixel_ratio) {
+    float dummy;
+    const auto start_dist = (x * in_size) / static_cast<double>(out_size);
```

Comment:
If it matters(e.g visible by profiler) then you can get rid of double divisions by sending the inverted value as a kernel parameter.